File: Linked-List/python/circularlinkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class CircularLinkedList:
    def __init__(self):
        self.head = None

    def append(self, value):
        new_node = Node(value)

        if self.head is None:
            self.head = new_node
            new_node.next = self.head
        else:
            temp = self.head
            while temp.next != self.head:
                temp = temp.next
            temp.next = new_node
            new_node.next = self.head

    def log_values(self):
        if self.head is None:
            print("Circular Linked List is empty.")
            return

        temp = self.head
        while True:
            print(temp.data, end=" ")
            temp = temp.next
            if temp == self.head:
                break
        print()


# Example usage:
my_list = CircularLinkedList()
my_list.append(1)
my_list.append(2)
my_list.append(3)

print("Circular Linked List values:")
my_list.log_values()

File: Linked-List/cpp/circularlinkedlist.cpp
#include <iostream>

using namespace std;

struct Node {
  int data;
  Node *next;
};

class CircularLinkedList {
private:
  Node *head;

public:
  CircularLinkedList() { head = nullptr; }

  void append(int value) {
    Node *newNode = new Node;
    newNode->data = value;
    newNode->next = nullptr;

    if (head == nullptr) {
      head = newNode;
      newNode->next = head;
    } else {
      Node *temp = head;
      while (temp->next != head) {
        temp = temp->next;
      }
      temp->next = newNode;
      newNode->next = head;
    }
  }

  void logValues() {
    if (head == nullptr) {
      cout << "Circular Linked List is empty." << endl;
      return;
    }

    Node *temp = head;
    do {
      cout << temp->data << " ";
      temp = temp->next;
    } while (temp != head);
    cout << endl;
  }
};

int main() {
  CircularLinkedList myList;
  myList.append(1);
  myList.append(2);
  myList.append(3);

  cout << "Circular Linked List values: ";
  myList.logValues();

  return 0;
}
